cli/napi/js_native_api.rs

```diff
@@ -0,0 +1,2266 @@
+  if !data.is_null() {
+    *data = get_array_buffer_ptr(value);
+  }
+  let value = v8::Uint8Array::new(&mut env.scope(), value, 0, len).unwrap();
+  let value: v8::Local<v8::Value> = value.into();
+  *result = transmute::<v8::Local<v8::Value>, napi_value>(value);
+  Ok(())
+}
+
+#[napi_sym::napi_sym]
+fn napi_create_buffer_copy(
+  env: *mut Env,
+  len: usize,
+  data: *mut u8,
+  result_data: *mut *mut u8,
+  result: *mut napi_value,
+) -> Result {
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+  let value = v8::ArrayBuffer::new(&mut env.scope(), len);
+  let ptr = get_array_buffer_ptr(value);
+  std::ptr::copy(data, ptr, len);
+  if !result_data.is_null() {
+    *result_data = ptr;
+  }
+  let value = v8::Uint8Array::new(&mut env.scope(), value, 0, len).unwrap();
+  let value: v8::Local<v8::Value> = value.into();
+  *result = transmute::<v8::Local<v8::Value>, napi_value>(value);
+  Ok(())
+}
+
+#[napi_sym::napi_sym]
+fn napi_coerce_to_bool(
+  env: *mut Env,
+  value: napi_value,
+  result: *mut napi_value,
+) -> Result {
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+  let value = transmute::<napi_value, v8::Local<v8::Value>>(value);
+  let coerced = value.to_boolean(&mut env.scope());
+  let value: v8::Local<v8::Value> = coerced.into();
+  *result = transmute::<v8::Local<v8::Value>, napi_value>(value);
+  Ok(())
+}
+
+#[napi_sym::napi_sym]
+fn napi_coerce_to_number(
+  env: *mut Env,
+  value: napi_value,
+  result: *mut napi_value,
+) -> Result {
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+  let value = transmute::<napi_value, v8::Local<v8::Value>>(value);
+  let coerced = value
+    .to_number(&mut env.scope())
+    .ok_or(Error::NumberExpected)?;
+  let value: v8::Local<v8::Value> = coerced.into();
+  *result = transmute::<v8::Local<v8::Value>, napi_value>(value);
+  Ok(())
+}
+
+#[napi_sym::napi_sym]
+fn napi_coerce_to_object(
+  env: *mut Env,
+  value: napi_value,
+  result: *mut napi_value,
+) -> Result {
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+  let value = transmute::<napi_value, v8::Local<v8::Value>>(value);
+  let coerced = value.to_object(&mut env.scope()).unwrap();
+  let value: v8::Local<v8::Value> = coerced.into();
+  *result = transmute::<v8::Local<v8::Value>, napi_value>(value);
+  Ok(())
+}
+
+#[napi_sym::napi_sym]
+fn napi_coerce_to_string(
+  env: *mut Env,
+  value: napi_value,
+  result: *mut napi_value,
+) -> Result {
+  let env: &mut Env = env.as_mut().ok_or(Error::InvalidArg)?;
+  let value = transmute::<napi_value, v8::Local<v8::Value>>(value);
+  let coerced = value.to_string(&mut env.scope()).unwrap();
+  let value: v8::Local<v8::Value> = coerced.into();
+  *result = transmute::<v8::Local<v8::Value>, napi_value>(value);
+  Ok(())
+}
+
+#[napi_sym::napi_sym]
+fn napi_create_dataview(
+  env: *mut Env,
+  len: usize,
+  data: *mut *mut u8,
+  byte_offset: usize,
+  result: *mut napi_value,
```

Comment:
Yikes, this ain't right. `data` should be a napi_value ArrayBuffer

Reply:
It seems that `data` is not used at all, that seems wrong too?